Sources/NIO/EventLoop.swift

```diff
@@ -586,7 +600,12 @@ extension EventLoop {
     /// - returns: a succeeded `EventLoopFuture`.
     @inlinable
     public func makeSucceededFuture<Success>(_ value: Success, file: StaticString = #file, line: UInt = #line) -> EventLoopFuture<Success> {
-        return EventLoopFuture<Success>(eventLoop: self, value: value, file: file, line: line)
+        if Success.self == Void.self {
+            // The as! will always succeed because we previously checked that Success.self == Void.self.
+            return self.makeSucceededVoidFuture() as! EventLoopFuture<Success>
```

Comment:
Why is this preferred over `func makeSucceededFuture(_ value: Void, file: StaticString = #file, line: UInt = #line) -> EventLoopFuture<Void>`?

Reply:
Because this one is cached.

Reply:
Oh, also because this one hits unspecialised generic code and doesn't potentially cause confusion in the type checker.

Reply:
For clarity, I meant why not:

```swift
func makeSucceededFuture(_ value: Void, file: StaticString = #file, line: UInt = #line) -> EventLoopFuture<Void> {
  return self.makeSucceededVoidFuture()
}
```

Anyway, "because this one hits unspecialised generic code and doesn't potentially cause confusion in the type checker" is what I was looking for, thanks!